src/cpp/library/random

```diff
@@ -0,0 +1,179 @@
+#pragma once
+
+#include <cstdint>
+
+extern int nondet_int();
+extern unsigned int nondet_uint();
+extern long nondet_long();
+extern unsigned long nondet_ulong();
+extern uint32_t nondet_uint32();
+extern uint64_t nondet_uint64();
+
+namespace nondet_util
+{
+// They are not part of the standard library and were added to simplify
+// the model.
+template <typename T>
+struct nondet_helper;
+
+template <>
+struct nondet_helper<int>
+{
+  static int get()
+  {
+    return nondet_int();
+  }
+};
+
+template <>
+struct nondet_helper<unsigned int>
+{
+  static unsigned int get()
+  {
+    return nondet_uint();
+  }
+};
+
+template <>
+struct nondet_helper<long>
+{
+  static long get()
+  {
+    return nondet_long();
+  }
+};
+
+template <>
+struct nondet_helper<uint64_t>
+{
+  static uint64_t get()
+  {
+    return nondet_uint64();
+  }
+};
+
+} // namespace nondet_util
+
+namespace std
+{
+
+class random_device
+{
+public:
+  using result_type = unsigned int;
+
+  random_device()
+  {
+  }
+
+  result_type operator()()
+  {
+    result_type val = nondet_uint();
+    __ESBMC_assume(val >= min() && val <= max());
+    return val;
+  }
+
+  static constexpr result_type min()
+  {
+    return 0;
+  }
+
+  static constexpr result_type max()
+  {
+    return 0x7FFFFFFF;
```

Comment:
Isn't this wrong?
https://en.cppreference.com/w/cpp/numeric/random/random_device/max.html
https://godbolt.org/z/7aa4fv86d

Reply:
@XLiZHI